topo/topo.go

```diff
@@ -229,20 +225,80 @@ func (m *Manager) Load(ctx context.Context) error {
 	return nil
 }
 
-// Topology gets the topology CRDs for the cluster.
-func (m *Manager) Topology(ctx context.Context) ([]topologyv1.Topology, error) {
+// TopologyResources gets the topology CRDs for the cluster.
+func (m *Manager) TopologyResources(ctx context.Context) ([]*topologyv1.Topology, error) {
 	topology, err := m.tClient.Topology(m.proto.Name).List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to get topology CRDs: %v", err)
 	}
-	return topology.Items, nil
+
+	items := make([]*topologyv1.Topology, len(topology.Items))
+	for i := range items {
+		items[i] = &topology.Items[i]
+	}
+
+	return items, nil
 }
 
 // Topology returns the topology protobuf.
 func (m *Manager) TopologyProto() *tpb.Topology {
 	return m.proto
 }
 
+func (m *Manager) TopologySpecs(ctx context.Context) ([]*topologyv1.Topology, error) {
+	nodeSpecs := map[string]*[]*topologyv1.Topology{}
+	topos := []*topologyv1.Topology{}
+
+	// get topology specs from all nodes
+	for _, n := range m.nodes {
+		log.Infof("Getting topology specs for node %s", n.Name())
+		specs, err := n.TopologySpecs(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch topology specs for node %s: %v", n.Name(), err)
+		}
+
+		log.Tracef("Topology specs for node %s: %+v", n.Name(), specs)
+		nodeSpecs[n.Name()] = &specs
+	}
+
+	// replace node name with pod name, for peer pod attribute in each link
+	for nodeName, specs := range nodeSpecs {
+		for _, spec := range *specs {
+			for l := range spec.Spec.Links {
+				link := &spec.Spec.Links[l]
+				peerSpecs, ok := nodeSpecs[link.PeerPod]
+				if !ok {
+					return nil, fmt.Errorf("specs do not exist for node %s", link.PeerPod)
+				}
+
+				foundPeer := false
+				for _, peerSpec := range *peerSpecs {
+					for _, peerLink := range peerSpec.Spec.Links {
```

Comment:
so if you refactor this into a helper func it will be easier to handle the found 

func replacePeer(link, peerSpecs) error {
  for _, peerSpec := range peerSpecs {
   ....
}

then you don't need the found and breaks well as many

}

Reply:
Thanks, that actually helped !
https://github.com/google/kne/pull/138/commits/db2ba56de37628c405180ae609ceb6ee706c4011